alternator/executor.cc

```diff
@@ -929,7 +929,7 @@ static future<executor::request_return_type> create_table_on_shard0(tracing::tra
             }
             sstring where_clause = "\"" + view_hash_key + "\" IS NOT NULL";
             if (!view_range_key.empty()) {
-                where_clause = where_clause + " AND \"" + view_hash_key + "\" IS NOT NULL";
+                where_clause = where_clause + " AND \"" + view_range_key + "\" IS NOT NULL";
```

Comment:
It appears you're absolutely right - the IS NOT NULL thing is only checked in the CQL front-end, so it was rather pointless to add it in the internal Alternator code - it would probably work without any IS NOT NULL but since we already add it, let's add it correctly so your patch is good.
 
You're right that the check in CQL has some weird corner-cases like "p is not null" isn't needed. I think part of that logic is based on a misunderstanding of the difference between null and empty strings, but you're right that this is orthogonal to this issue. I'll think about that other issue later.